torch/_inductor/codegen/cpp.py

```diff
@@ -837,15 +837,15 @@ class CppOverrides(OpOverrides):
 
     @staticmethod
     def add(a, b):
-        return f"decltype({a})({a} + {b})"
+        return f"decltype({a})({a} + {b}); static_assert(std::is_same<decltype({a}), decltype({b})>::value)"
```

Comment:
This can't be correct, as you added this casting to fix the default casting of the `+` operator in some cases. This can only happen if the type of a and b are different.

I think what you are looking for is `is_floating_point_v`, I guess

Reply:
Thanks. I'm considering to model the cse variable with `CppCSEVariable` and check the types from python. Will do that in a separate PR. I simplified the UT per your suggestion in this PR (forgot to do that previously).

Reply:
> This can't be correct, as you added this casting to fix the default casting of the + operator in some cases. This can only happen if the type of a and b are different.

The original problem here is that `uint8 + uint8 -> int` in C++, or indeed any integer type less than `int` results in an `int`. That said, I still wouldn't be surprised if this `static_assert` fails so the point still stands.

Reply:
The expected dtypes are `uint8 + uint8 -> uint8` which aligned with the generated C++ code with explicit type-cast on the result. If we know the dtypes from the python codegen, we can make sure the generated code has the expected dtypes for input and output without the need of adding `static_assert` here. So I'd prefer check that from python codegen, not from inside the generated C++ code which sounds more complicated.